tensorflow_addons/rnn/tests/cell_test.py

```diff
@@ -410,3 +410,139 @@ def test_configs_layernorm():
     cell2 = LayerNormSimpleRNNCell(**config1)
     config2 = cell2.get_config()
     assert config1 == config2
+
+
+def test_base_esn():
+    units = 3
+    expected_output = np.array(
+        [[2.77, 2.77, 2.77], [4.77, 4.77, 4.77], [6.77, 6.77, 6.77],], dtype=np.float32,
+    )
+
+    const_initializer = tf.constant_initializer(0.5)
+    cell = rnn_cell.ESNCell(
+        units=units,
+        connectivity=1,
+        leaky=1,
+        spectral_radius=0.9,
+        use_norm2=True,
+        use_bias=True,
+        activation=None,
+        kernel_initializer=const_initializer,
+        recurrent_initializer=const_initializer,
+        bias_initializer=const_initializer,
+    )
+
+    inputs = tf.constant(
+        np.array(
+            [[1.0, 1.0, 1.0, 1.0], [2.0, 2.0, 2.0, 2.0], [3.0, 3.0, 3.0, 3.0]],
+            dtype=np.float32,
+        ),
+        dtype=tf.float32,
+    )
+    state_value = tf.constant(
+        0.3 * np.ones((units, units), dtype=np.float32), dtype=tf.float32
+    )
+    init_state = [state_value, state_value]
+    output, state = cell(inputs, init_state)
+
+    np.testing.assert_allclose(output, expected_output, 1e-5)
+    np.testing.assert_allclose(state, expected_output, 1e-5)
+
+
+def test_esn_echo_state_property_eig():
+    use_norm2 = False
+    units = 3
+    cell = rnn_cell.ESNCell(
+        units=units,
+        use_norm2=use_norm2,
+        recurrent_initializer="ones",
+        connectivity=1.0,
+    )
+    cell.build((3, 3))
+    recurrent_weights = tf.constant(cell.get_weights()[0], dtype=tf.float32)
+    max_eig = tf.reduce_max(tf.abs(tf.linalg.eig(recurrent_weights)[0]))
+    assert max_eig < 1, "max(eig(W)) < 1"
+
+
+def test_esn_echo_state_property_norm2():
+    use_norm2 = True
+    units = 3
+    cell = rnn_cell.ESNCell(
+        units=units, use_norm2=use_norm2, recurrent_initializer="ones", connectivity=1.0
+    )
+    cell.build((3, 3))
+    recurrent_weights = tf.constant(cell.get_weights()[0])
+    max_eig = tf.reduce_max(tf.abs(tf.linalg.eig(recurrent_weights)[0]))
+    assert max_eig < 1, "max(eig(W)) < 1"
+
+
+def test_esn_connectivity():
+    units = 1000
+    connectivity = 0.5
+    cell = rnn_cell.ESNCell(
+        units=units,
+        connectivity=connectivity,
+        use_norm2=True,
+        recurrent_initializer="ones",
+    )
+    cell.build((3, 3))
+    recurrent_weights = tf.constant(cell.get_weights()[0])
+    num_non_zero = tf.math.count_nonzero(recurrent_weights)
+    actual_connectivity = tf.divide(num_non_zero, units ** 2)
+    np.testing.assert_allclose(
+        np.asarray([actual_connectivity]), np.asanyarray([connectivity]), 1e-2
+    )
+
+
+def test_esn_keras_rnn():
```

Comment:
Please add another test case for a e2e test with keras model and compile/fit.

Reply:
Added!